index.js

```diff
@@ -1,33 +1,57 @@
 const fetch = require('node-fetch'); // Automatically excluded in browser bundles
 
-async function api(endpoint, token) {
-	token = token ? `&access_token=${token}` : '';
-	const response = await fetch(`https://api.github.com/repos/${endpoint}${token}`);
+// Matches '/<user>/<repo>/tree/<ref>/<dir>'
+const urlParserRegex = /^[/]([^/]+)[/]([^/]+)[/]tree[/]([^/]+)[/](.*)/;
+
+function parseGithubURL(url) {
+	const res = {};
+	const parsedUrl = new URL(url);
+	[, res.user, res.repository, res.ref, res.directory] = urlParserRegex.exec(parsedUrl.pathname) || [];
+	if (typeof res.directory !== 'string') {
+		// eslint-disable-next-line unicorn/prefer-type-error
+		throw new Error('Unable to parse GitHub URL');
+	}
+
+	if (parsedUrl.hostname === 'github.com') {
+		res.api = 'https://api.github.com';
+	} else {
+		res.api = `https://${parsedUrl.host}/api/v3`;
+	}
+
+	// TODO: Validate
+	return res;
+}
+
+async function githubApi(api, endpoint, token) {
+	const response = await fetch(`${api}/repos/${endpoint}`, {
+		headers: {
+			Authorization: `Bearer ${token}`
+		}
+	});
 	return response.json();
 }
 
 // Great for downloads with few sub directories on big repos
 // Cons: many requests if the repo has a lot of nested dirs
 async function viaContentsApi({
-	user,
-	repository,
-	ref = 'HEAD',
-	directory,
+	resource: res,
```

Comment:
I think this could just be called `resource`?

```suggestion
	resource,
```

Reply:
It makes the references to that object below much more uselessly long. So I preferred to alias it. Let me know if you want me to change it.

Reply:
It's `resource` in the first place, we have two names for the same thing. Also this rule will be enabled soon: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prevent-abbreviations.md

Reply:
Fixed!